pkg/client/cache/controller.go

```diff
@@ -25,6 +25,8 @@ import (
 	"k8s.io/kubernetes/pkg/util/wait"
 )
 
+const stopMarker ExplicitKey = "stopMe"
```

Comment:
Let me work on a different approach if we want to work that way. It will require another goroutine and a flag on top of the error.
I was trying to go with a "clean" approach which required little to no change.